crates/sim2h/src/job/listen.rs

```diff
@@ -0,0 +1,36 @@
+use crate::*;
+
+/// a job that manages the sim2h listening socket connection
+/// every iteration will `accept()` a single pending connection
+pub(crate) struct ListenJob {
+    listen: TcpWssServer,
+    wss_send: crossbeam_channel::Sender<TcpWss>,
+}
+
+impl ListenJob {
+    pub(crate) fn new(listen: TcpWssServer, wss_send: crossbeam_channel::Sender<TcpWss>) -> Self {
+        Self { listen, wss_send }
+    }
+
+    fn run(&mut self) -> JobContinue {
+        match self.listen.accept() {
```

Comment:
does this need the same check of `self.cont` as in the `ConnectionJob` run function?

Reply:
Right now the listening socket just lives for the whole life of the process (or panic!s on error : ) - we can totally make this more robust in the future if we have need to start / stop listening on network interfaces at run-time.

Reply:
Yah I saw that panic.  What kind of error could there be?  I.e. what happens to the listening socket if the user changes the network interface?   I guess for the sim2h_server this shouldn't happen, so we can ignore.